service/host.go

```diff
@@ -130,12 +134,21 @@ func (uis *UIServer) modifyHost(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		err := h.SetStatus(newStatus, u.Id, "")
-		if err != nil {
-			uis.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "Error updating host"))
-			return
+		var msg flashMessage
```

Comment:
Do you mind doing a bit of refactoring to make this testable, then adding tests? The code in our UI routes has historically been written in a way that's not unit testable, but we're changing them and adding tests.

I would take the logic from lines 123 to the end and put it in the model/host package, and leave any HTTP-related stuff in this function

Reply:
How would you recommend error handling? Right now the logic returns a mix of HTTP 400 and 500 errors, but I feel like just encapsulating all of those as just Go `errors` and letting the route return them as a general "HTTP 500 something went wrong", like in lines 148-148, wouldn't be super useful. I also thought about returning an `error` in the refactored function to the route, and then having the route check the error string to figure out the appropriate HTTP error code to pass back to the client, but I'm not sure if there's a better way to do that than just straight up string comparison

Reply:
My strategy has been the following:
Database errors = 500.
Host not found = 400. 
Auth error = 404 or 403 depending on whether or not hiding the existence of something is a good idea.
500 everything else.